x-pack/solutions/security/plugins/security_solution/server/endpoint/services/workflow_insights/index.test.ts

```diff
@@ -491,4 +496,112 @@ describe('SecurityWorkflowInsightsService', () => {
       });
     });
   });
+
+  describe('agent space-awareness and lifecycle hooks', () => {
+    let request: KibanaRequest<unknown, unknown, DefendInsightsGetRequestQuery>;
+
+    const setupWithMockFleet = (options: {
+      featureEnabled: boolean;
+      ensureInCurrentSpaceMock?: jest.Mock;
+    }) => {
+      // @ts-expect-error write to readonly property
+      mockEndpointAppContextService.experimentalFeatures.endpointManagementSpaceAwarenessEnabled =
+        options.featureEnabled;
+
+      const ensureInCurrentSpace = options.ensureInCurrentSpaceMock ?? jest.fn();
+
+      const fleetServices = {
```

Comment:
FYI: The internal Fleet Services module already has a Mocked version of it for use in tests ([see here](https://github.com/elastic/kibana/blob/7f9ff2dbfdb4151bf9b241010284218cab1cfc84/x-pack/solutions/security/plugins/security_solution/server/endpoint/services/fleet/endpoint_fleet_services_factory.mocks.ts#L40)) and it is already used in the endpoint services mock created by `createMockEndpointAppContext()` ([see here](https://github.com/elastic/kibana/blob/5ae36a5a8b6b9bac65290f5db7b9833f92952e65/x-pack/solutions/security/plugins/security_solution/server/endpoint/mocks/mocks.ts#L108-L110)).

Reply:
Yeah, thanks for pointing this out. This test went through a couple iterations and a lot was left over, cleaned it up a little in a717979e0ec1a0893feb2eff095a8224e4679672. Thanks!